### PR TITLE
Replace font_assets with rack-cors

### PR DESCRIPTION
**Pull request by akostadinov**
**What this PR does / why we need it**:

font_assets is obsolete and unsupported. rack-cors open possibilities for a richer CORS support when needed.

**Which issue(s) this PR fixes** 

THREESCALE-7515

**Verification steps** 

For font_assets behavior see https://github.com/3scale/porta/pull/2581#issuecomment-911502552

This is GET request with rack-cors:
```
🐚 curl -v -H "Origin: localhost" -H "Access-Control-Request-Method: GET" http://provider-admin.3scale.localhost:3000/packs/media/overpass-webfont/overpass-light-c97e1959.woff2
*   Trying ::1:3000...
* connect to ::1 port 3000 failed: Connection refused
*   Trying 127.0.0.1:3000...
* Connected to provider-admin.3scale.localhost (127.0.0.1) port 3000 (#0)
> GET /packs/media/overpass-webfont/overpass-light-c97e1959.woff2 HTTP/1.1
> Host: provider-admin.3scale.localhost:3000
> User-Agent: curl/7.76.1
> Accept: */*
> Origin: localhost
> Access-Control-Request-Method: GET
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET
< Access-Control-Max-Age: 3628800
< Last-Modified: Wed, 01 Sep 2021 08:28:04 GMT
< Content-Type: application/font-woff2
< Cache-Control: public, max-age=31557600
< Content-Length: 34472
< Vary: Accept-Encoding, Origin
< X-Rack-CORS: hit
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Connection: keep-alive
< Server: thin
< 
...
```

For OPTIONS requests with rack-cors:
```
🐚 curl -v -X OPTIONS -H "Origin: localhost" -H "Access-Control-Request-Method: GET" http://provider-admin.3scale.localhost:3000/packs/media/overpass-webfont/overpass-light-c97e1959.woff2
*   Trying ::1:3000...
* connect to ::1 port 3000 failed: Connection refused
*   Trying 127.0.0.1:3000...
* Connected to provider-admin.3scale.localhost (127.0.0.1) port 3000 (#0)
> OPTIONS /packs/media/overpass-webfont/overpass-light-c97e1959.woff2 HTTP/1.1
> Host: provider-admin.3scale.localhost:3000
> User-Agent: curl/7.76.1
> Accept: */*
> Origin: localhost
> Access-Control-Request-Method: GET
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET
< Access-Control-Max-Age: 3628800
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Connection: close
< Server: thin
< 
* Closing connection 0
```

### CODE REVIEW

File: config/examples/cors.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+cors: &default
+  enabled: false
+  allow:
+  - origins: '*'
+    resources:
+    - !ruby/regexp /\.(?:woff2?|otf|ttf|svg|eot)$/
+    headers: 'x-requested-with'
+    methods: :get
+    max_age: 3628800
+    credentials: false
```
Comment (akostadinov) [inline config/examples/cors.yml:10]: This is something I changed from initial implementation. This is an array of origin/resources/other_options bundles. Reading on the web it seems like sometimes multiple allow configurations are desirable. If you think that only a single origin/resources/other_options is well enough, then I can revert.

Actually rack-cors allows in a single `allow` rule to have multiple resources with different other_options (headers, methods, vary, etc.) but I thought this is too fine grained.

The example `cors.yml` replicates what we had with `font_assets`.

File: lib/three_scale/middleware/cors.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    private
+
+    def set_config
+      rules = config.allow.presence || []
+
+      rules.map(&:symbolize_keys).each do |rule|
+        origins = rule[:origins].presence || '*'
+        resources = rule[:resources].presence || '*'
+        methods = rule[:methods].presence || :get
```
Comment (akostadinov) [inline lib/three_scale/middleware/cors.rb:37]: In initial implementation default was to allow all methods. I've put a more conservative default here of only `GET` requests.

**Comment by akostadinov** [comment]
There are some codeclimate complaints. A number are about original code, a number about `set_config` method. IMO these suggested changes will likely complicate things more than do any good.

Please see if functionality if fine with you. And which codeclimate complaints are desirable to be fixed.

**Review by guicassolato — APPROVED** [review]
Nice job!

**Review by hallelujah — APPROVED** [review]
Awesome :rocket:

**Pull request merged**